staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -418,6 +418,13 @@ const (
 	DeletePropagationForeground DeletionPropagation = "Foreground"
 )
 
+// Possible values for DryRun flag.
+const (
+	// DryRunAll means that the entire chain should be verified: admission
+	// chain, validation, merging, everything.
+	DryRunAll = "All"
```

Comment:
if this is what we call this, let's make sure we agree that any new stages in the api pipeline in the future will get included in this mode, so that we never ever ever have to have a repeat of the "all does not mean all" issue. cc @smarterclayton

Reply:
I don't think this is referenced anywhere yet. If we want to drop this from this PR to get the signature/plumbing changes in, and continue discussion of the semantics of the parameter, that would be fine with me